### PR TITLE
Feature/csl1026 git

### PR DESCRIPTION
**Pull request by 7sedam7**
Store blocks in subdirectories, like git does.

### CODE REVIEW

File: src/Pos/DB/Block.hs [hunk 1]
```diff
@@ -247,13 +248,19 @@ deleteData fp = (liftIO $ removeFile fp) `catch` handle
         | isDoesNotExistError e = pure ()
         | otherwise = throwM e
 
+dirDataPath :: MonadDB m => HeaderHash -> m FilePath
+dirDataPath (formatToString hashHexF -> fn) = gitDirDataPath fn
```
Comment (volhovM) [inline src/Pos/DB/Block.hs]: I guess it'd be more obvious to store them by epoch index. Why did you go with hash appproach?

File: src/Pos/DB/Block.hs [hunk 2]
```diff
@@ -247,13 +248,19 @@ deleteData fp = (liftIO $ removeFile fp) `catch` handle
...
+
 blockDataPath :: MonadDB m => HeaderHash -> m FilePath
 blockDataPath (formatToString (hashHexF%".block") -> fn) =
-    getNodeDBs <&> \dbs -> dbs ^. blockDataDir </> fn
+    gitDirDataPath fn <&> (</> drop 2 fn)
```
Comment (volhovM) [inline src/Pos/DB/Block.hs]: Why don't you use full `fn` here (and below)?

File: src/Pos/DB/Block.hs (hunk 1 above)
Comment (neongreen) [inline src/Pos/DB/Block.hs]: I think this was discussed in #vatnik (or somewhere else?) and we decided to go with the hash approach, but I don't remember why.

File: src/Pos/DB/Block.hs (hunk 2 above)
Comment (gromakovsky) [inline src/Pos/DB/Block.hs]: Because first two characters form directory name.

**Pull request merged**